File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCount{0};
        int start {-1};
        for(int i=0;i<nums.size();i++){
            if(start == -1 && nums[i] == 1){
                start = 1;
                continue;
            }
            if(nums[i] == 1 && start != -1) start++;
            if(nums[i] == 0 && start != -1){
                maxCount = maxCount>start?maxCount:start;
                start = -1;
            }
        }
        return maxCount>start?maxCount:start;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans=0
        temp=0
        for i in nums:
            if i==1:
                temp+=1
            else:
                ans=max(ans,temp)
                temp=0
        ans=max(temp,ans)
        return ans